qutip/tests/core/test_ptrace.py

```diff
@@ -130,3 +130,20 @@ def test_ptrace_operator(dtype, sel):
         qutip.rand_dm(2), qutip.thermal_dm(10, 1), qutip.rand_unitary(3),
     ).to(dtype)
     assert A.ptrace(sel) == expected(A, sel)
+
+@pytest.mark.parametrize('dims, sel',
+                         [
+                             ([5, 2, 3], [2, 1]),
+                             ([5, 2, 3], [0, 2]),
+                             ([5, 2, 3], [0, 1]),
+                             ([2]*6, [3, 2]),
+                             ([2]*6, [0, 2]),
+                             ([2]*6, [0, 1]),
```

Comment:
These pattern seems redundant. 
Why not also vary the number of kept indices?

Reply:
I had just copied an existing test without looking too much at it :D I now added some more relevant test cases